sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -641,12 +641,19 @@ def __init__(
   def estimate_size(self):
     bq = bigquery_tools.BigQueryWrapper()
     if self.table_reference is not None:
+      table_ref = self.table_reference
+      if (isinstance(self.table_reference, vp.ValueProvider) and
+          self.table_reference.is_accessible()):
+        table_ref = bigquery_tools.parse_table_reference(
+            self.table_reference.get(), self.dataset, self.project)
+      elif isinstance(self.table_reference, vp.ValueProvider):
+        # Size estimation is best effort. We return 0 as we have no
+        # access to the table that we're querying.
+        return 0
       table = bq.get_table(
-          self.table_reference.projectId,
-          self.table_reference.datasetId,
-          self.table_reference.tableId)
+          table_ref.projectId, table_ref.datasetId, table_ref.tableId)
       return int(table.numBytes)
-    else:
+    elif self.query is not None and self.query.is_accessible():
```

Comment:
Shouldn't we support ValueProvider for queries as well ?

Reply:
It is supported now. In this case, the `self.query` parameter is wrapped by a ValueProvider already. This does not work for `self.table` because it may be a non-string.